controllers/hetznerbaremetalhost_controller.go

```diff
@@ -188,6 +196,15 @@ func (r *HetznerBareMetalHostReconciler) reconcileSelectedStates(ctx context.Con
 		// Handle StateDeleting
 	case infrav1.StateDeleting:
 		if !utils.StringInList(bmHost.Finalizers, infrav1.BareMetalHostFinalizer) {
+			conditions.MarkFalse(
```

Comment:
I think you can remove this. Right now it doesn't have any effect (see that you don't update the host after this). You could, if anything, add it to the code below, but also there you remove the finalizer already. After the finalizer is removed, the object will be deleted very quickly. Therefore, we probably don't need a condition here at all.